src/runtime.rs

```diff
@@ -304,8 +302,22 @@ impl RunState {
             }
             // putsp
             0x24 => {
-                // TODO: impl putsp
-                todo!("TODO: putsp can be put off until someone needs it")
+                let mut addr = *self.reg(0);
+                loop {
+                    let chr_raw = *self.mem(addr);
+                    let chr_high = ((chr_raw >> 8) & 0xFF) as u8 as char;
```

Comment:
`mem` returns a u16 so not sure what the need is to `& xFF` if it has been shifted by 8